### PR TITLE
[Impeller] dont use blend shader for simple drawAtlas calls.

### PR DESCRIPTION
**Pull request by jonahwilliams**
When making a drawAtlas call with no colors, use the faster shader that does not apply blending.

Now that we're switched to experimental canvas, we can remove the copying of the display list data. The draw can directly refer to pointers from the display list structure, as the draw is processed during DL dispatching.

### CODE REVIEW

File: impeller/display_list/dl_atlas_geometry.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#include "display_list/dl_color.h"
+#include "display_list/image/dl_image.h"
+#include "impeller/core/sampler_descriptor.h"
+#include "impeller/entity/contents/atlas_contents.h"
+#include "impeller/geometry/color.h"
+#include "include/core/SkRSXform.h"
+
+namespace impeller {
+
+/// @brief A geometry that is created from a DlVertices object.
```
Comment (jonahwilliams) [inline impeller/display_list/dl_atlas_geometry.h:17]: hmmm, not quite

**Comment by jonahwilliams** [comment]
I will add some unit tests of the vertex computations.

**Review by jtmcdole — APPROVED** [review]
Would be nice to have some test coverage for the new files; but LGTM

**Comment by jonahwilliams** [comment]
Unit tests adde

**Pull request merged**